xcdat/bounds.py

```diff
@@ -428,239 +467,238 @@ def _create_bounds(self, axis: CFAxisKey, coord_var: xr.DataArray) -> xr.DataArr
-    """Creates time bounds for each timestep with the start and end of the year.
+        Raises
+        ------
+        TypeError
+            If time coordinates are not composed of ``cftime.datetime`` object.
+        """
 
-    Bounds for each timestep correspond to Jan. 1 00:00:00 of the year of the
-    timestep and Jan. 1 00:00:00 of the subsequent year.
+        freq = _infer_freq(time) if freq is None else freq  # type: ignore
+
+        if freq == "year":
+            time_bnds = self._create_yearly_time_bounds(time)
+        elif freq == "month":
+            time_bnds = self._create_monthly_time_bounds(time)
+        elif freq == "day":
+            time_bnds = self._create_daily_time_bounds(time)
+        elif freq == "hour":
+            # Determine the daily frequency for generating time  bounds.
+            if daily_freq is None:
+                diff = time.values[1] - time.values[0]
+                hrs = diff.seconds / 3600
+                daily_freq = int(24 / hrs)  # type: ignore
+
+            time_bnds = self._create_daily_time_bounds(time, freq=daily_freq)  # type: ignore
+
+        # Create the bounds data array
+        da_time_bnds = xr.DataArray(
+            name=f"{time.name}_bnds",
+            data=time_bnds,
+            coords={time.name: time},
+            dims=[*time.dims, "bnds"],
+            attrs={"xcdat_bounds": "True"},
+        )
 
-    Parameters
-    ----------
-    time : xr.DataArray
-        The temporal coordinate variable for the axis.
+        return da_time_bnds
 
-    Returns
-    -------
-    xr.DataArray
-        The monthly time bounds array.
-    """
-    # Get the calendar type from the time coordinates to determine the `cftime`
-    # object to represent bounds values.
-    calendar = time.encoding["calendar"]
-    cf_obj = get_date_type(calendar)
+    def _create_yearly_time_bounds(self, time: xr.DataArray) -> List[cftime.datetime]:
+        """Creates time bounds for each timestep with the start and end of the year.
+
+        Bounds for each timestep correspond to Jan. 1 00:00:00 of the year of the
+        timestep and Jan. 1 00:00:00 of the subsequent year.
 
-    # Loop over the time values to generate a bounds DataArray.
-    time_bnds = []
-    for step in time.values:
-        year = step.year
+        Parameters
+        ----------
+        time : xr.DataArray
+            The temporal coordinate variable for the axis.
 
-        l_bnd = cf_obj(year, 1, 1, 0, 0)
-        u_bnd = cf_obj(year + 1, 1, 1, 0, 0)
+        Returns
+        -------
+        List[cftime.datetime]
+            A list of time bound values.
+        """
+        # Get the calendar type from the time coordinates to determine the `cftime`
+        # object to represent bounds values.
+        calendar = time.encoding["calendar"]
+        cf_obj = get_date_type(calendar)
 
-        time_bnds.append([l_bnd, u_bnd])
+        time_bnds: List[cftime.datetime] = []
```

Comment:
FIXME: We should detect if the input time coordinates are `datetime` or `cftime` and keep the time bounds the same object type.

Reply:
This is fixed in the latest commit.